### PR TITLE
[Bugfix] Druid `run_query` dimensions part 3 + Unit tests

### PR DESCRIPTION
**Pull request by Mogball**
https://github.com/apache/incubator-superset/pull/3920#issuecomment-347662484

Hopefully this'll be the last one. Added some unit tests to try to save some face 🥇

### CODE REVIEW

File: superset/connectors/druid/models.py [hunk 1]
```diff
@@ -898,7 +898,7 @@ def run_query(  # noqa / druid
             row_limit=None,
             inner_from_dttm=None, inner_to_dttm=None,
             orderby=None,
-            extras=None,  # noqa
+            extras={},  # noqa
```
Comment (mistercrunch) [inline superset/connectors/druid/models.py]: this is a python gotcha:
http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments,
just set it to `None` here and go with `extras = extras or {}` in the function body

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset/connectors/druid/models.py (hunk 1 above)
Comment (Mogball) [reply superset/connectors/druid/models.py]: 👍

**Review by mistercrunch — APPROVED** [review]

**Comment by ghulands** [comment]
Just seeing if you tested this with a time series chart too? That too was still broken after #3920.

**Comment by Mogball** [comment]
Yes I checked it with that. The test case is the one where `timeseries_limit` is specified

**Comment by ghulands** [comment]
I just tested this and it fails for the time series charts. I'm using a column in the group by having the same setup as I reported in #3943

**Comment by Mogball** [comment]
Looks like it's an issue with `_add_filter_from_pre_query_data`, which doesn't properly handle dimensions specs (and is called for timeseries). I think that's a bit outside the scope of this fix. And I'm not really familiar with PyDruid. I'll look into it though.

**Pull request merged**